modules/reactor/include/meshgenerators/PolygonMeshGeneratorBase.h

```diff
@@ -147,17 +150,140 @@ class PolygonMeshGeneratorBase : public MeshGenerator
+                    const multiBdryLayerParams & duct_outer_boundary_layer_params,
+                    const Real primary_side_length,
+                    const Real secondary_side_length,
+                    const unsigned int num_sectors_per_side,
+                    const unsigned int background_intervals,
+                    const Real background_radial_bias,
+                    const singleBdryLayerParams & background_inner_boundary_layer_params,
+                    const singleBdryLayerParams & background_outer_boundary_layer_params,
+                    dof_id_type & node_id_background_meta,
+                    const Real azimuthal_angle,
+                    const std::vector<Real> azimuthal_tangent,
+                    const unsigned int side_index,
+                    const bool quad_center_elements,
+                    const Real rotation_angle);
+
+  /**
+   * Generates a mesh of a slice, which is the foundation of both buildGeneralSlice and
+   * buildSimpleSlice.
+   * @param ring_radii radii of the ring regions
+   * @param ring_layers numbers of radial intervals of the ring regions
+   * @param ring_radial_biases values used for radial meshing biasing in ring regions
+   * @param ring_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the ring regions
+   * @param ring_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the ring regions
+   * @param ducts_center_dist distance parameters of the duct regions
+   * @param ducts_layers numbers of radial intervals of the duct regions
+   * @param duct_radial_biases values used for radial meshing biasing in duct regions
+   * @param duct_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the duct regions
+   * @param duct_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the duct regions
+   * @param pitch twice of the length of the first side times cosine of the azimuthal angle
+   * @param num_sectors_per_side number of azimuthal intervals
+   * @param background_intervals number of radial intervals of the background region
+   * @param background_radial_bias value used for radial meshing biasing in background region
+   * @param background_inner_boundary_layer_params width, radial sectors, and growth factor of the
+   * inner boundary layer of the background region
+   * @param background_outer_boundary_layer_params width, radial sectors, and growth factor of the
+   * outer boundary layer of the background region
+   * @param node_id_background_meta pointer to the first node's id of the background region
+   * @param virtual_side_number 360.0 over the azimuthal angle of the slice (happens to be number of
+   * sides of the polygon if a polygon is to be generated)
```

Comment:
when is a polygon not generated? For the circle regions?

Reply:
Should be "regular polygon" here.